### PR TITLE
 Fix FP S1472 (call-argument-line): Ignore type parameters 

### LINKED ISSUE #3705: Update S1472 (`call-argument-line`): cover scope of ESLint's `no-unexpected-multiline`
https://eslint.org/docs/latest/rules/no-unexpected-multiline

### PR DESCRIPTION
**Pull request by francoismora**
Fixes #3705

All the 14 [issues](https://peach.sonarsource.com/issues?resolved=false&rules=javascript%3AS1472%2Ctypescript%3AS1472) found by this rule in peach are false-positive due to a type parameter spanning multiple lines. The PR uses the type parameter when defined to determine the line where the arguments should start.

### CODE REVIEW

File: tests/linting/eslint/rules/call-argument-line.test.ts [hunk 1]
```diff
@@ -20,7 +20,10 @@
 import { RuleTester } from 'eslint';
 import { rule } from 'linting/eslint/rules/call-argument-line';
 
-const ruleTester = new RuleTester({ parserOptions: { ecmaVersion: 2018 } });
+const ruleTester = new RuleTester({
+  parser: require.resolve('@typescript-eslint/parser'),
+  parserOptions: { ecmaVersion: 2018 },
+});
```
Comment (yassin-kammoun-sonarsource) [inline tests/linting/eslint/rules/call-argument-line.test.ts:26]: I think you can use instead either `JavaScriptRuleTester` or `TypeScriptRuleTester` from `tests/tools/tests` which provides a wrapper for ESLint's rule tester using TypeScript ESLint parser.

**Review by yassin-kammoun-sonarsource — APPROVED** [review]
Left a minor suggestion, otherwise looks good to me.

File: tests/linting/eslint/rules/call-argument-line.test.ts (hunk 1 above)
Comment (francoismora) [reply tests/linting/eslint/rules/call-argument-line.test.ts]: Thanks for the tip. I was not aware of these helper classes. Fixed.

Comment (yassin-kammoun-sonarsource) [reply tests/linting/eslint/rules/call-argument-line.test.ts]: 👍

**Pull request merged**